src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs

```diff
@@ -153,7 +153,9 @@ private void GenerateCompilandsAndLog(BinaryAnalyzerContext context, List<Object
             // are out of policy: {1} 
             context.Logger.Log(this,
                 RuleUtilities.BuildResult(failureLevel, context, null,
-                nameof(RuleResources.BA2004_Warning_Native),
+                failureLevel == FailureLevel.Error
+                    ? nameof(RuleResources.BA2004_Error_Native_WithBadCompilands)

```

Comment:
>BA2004_Error_Native_WithBadCompilands [](start = 43, length = 37)

I wouldn't actually do this. Our standard is for the code to display, as a comment *every* user-facing string when one is returned. This helps with code readability. So instead, I'd suggest using a conditional here and creating two code comments that provide the text for these string resources. Note that the names of the strings don't describe the differentiation (which is between static library or not, suggest these:

BA2004_Error_NativeWithInsecureDirectCompilands
BA2004_Warning_NativeWithInsecureStaticLibraryCompilands